packages/svelte/src/compiler/phases/1-parse/state/element.js

```diff
@@ -194,8 +194,12 @@ export default function tag(parser) {
 	/** @type {Set<string>} */
 	const unique_names = new Set();
 
+	const current = parser.current();
+	const is_top_level_script_or_style =
+		(name === 'script' || name === 'style') && current.type === 'Root';
+
 	let attribute;
-	while ((attribute = read_attribute(parser, unique_names))) {
+	while ((attribute = read_attribute(parser, unique_names, is_top_level_script_or_style))) {
```

Comment:
what if rather than passing `is_static` around, and branching in different places, we separated the static attribute handling into its own function? #9505